ci/test-dev-context-only-utils.sh

```diff
@@ -2,4 +2,5 @@
 
 set -eo pipefail
 
-scripts/check-dev-context-only-utils.sh "$@"
+scripts/check-dev-context-only-utils.sh check-all-targets "$@"
+scripts/check-dev-context-only-utils.sh check-bins-and-lib "$@"
```

Comment:
thanks to the new `--partition` flag, i've changed to shard both modes `check-bin` (now called `check-bins-and-lib`) and `check-all-targets` more evenly.